### PR TITLE
checker(dm): make precheck warn/error limit public

### LINKED ISSUE #10850: DM precheck fail: `grpc: received message larger than max`
### What did you do?

Use dm on cloud to import

### What did you expect to see?

precheck success

### What did you see instead?

The precheck of DM task was failed by internal error. 
```
error="grpc: received message larger than max (6069063 vs. 4194304)"
```

The returned message of grpc is too long and confusing to users.

### Versions of the cluster

Cluster version: 
```
v7.5.0
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #10850

#### Tests 

 - Unit test

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]

**Review by okJiang — COMMENTED** [review]
I will review later

File: dm/checker/check_test.go [hunk 1]
```diff
@@ -173,6 +173,44 @@ func TestReplicationPrivilegeChecking(t *testing.T) {
...
+	mock = initMockDB(t)
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+
+	result, err = RunCheckOnConfigs(context.Background(), cfgs, false, 0, 1)
+	require.NoError(t, err)
+	require.EqualValues(t, 2, result.Summary.Failed)
+	require.Len(t, result.Results, 1)
```
Comment (okJiang) [inline dm/checker/check_test.go:211]: Can you test another case? multi errors contained by one result, and they are limited by `input:limit`.

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: dm/checker/check_test.go (hunk 1 above)
Comment (lance6716) [reply dm/checker/check_test.go:211]: https://github.com/pingcap/tiflow/pull/11061/commits/4ba16262fbfd2d54c9612031ad2029ae8f728374

**Review by okJiang — APPROVED** [review]

**Pull request merged**